packages/ui-components/src/components/Chart/Legend.js

```diff
@@ -4,36 +4,74 @@
  */
 
 import React from 'react';
-import { Legend as LegendComponent } from 'recharts';
-import PropTypes from 'prop-types';
-
-export const Legend = ({ chartConfig, onClick, getIsActiveKey, isExporting }) => {
-  const formatLegend = (value, { color }) => {
-    const isActive = getIsActiveKey(value);
-    return (
-      <span style={{ color, textDecoration: isActive ? '' : 'line-through' }}>
-        {chartConfig[value].label || value}
-      </span>
-    );
-  };
-
-  return (
-    <LegendComponent
-      onClick={onClick}
-      formatter={formatLegend}
-      verticalAlign={isExporting ? 'top' : 'bottom'}
-      wrapperStyle={isExporting ? { top: '-20px' } : {}}
-    />
-  );
-};
+import styled from 'styled-components';
+import MuiButton from '@material-ui/core/Button';
 
-Legend.propTypes = {
-  chartConfig: PropTypes.object.isRequired,
-  onClick: PropTypes.func.isRequired,
-  getIsActiveKey: PropTypes.func.isRequired,
-  isExporting: PropTypes.bool,
-};
+const LegendContainer = styled.div`
+  display: flex;
+  flex-wrap: wrap;
+  padding: 0 0 2rem 0;
+`;
+
+const LegendItem = styled(MuiButton)`
+  margin-right: 1.2rem;
+  font-size: 0.75rem;
+
+  .MuiButton-label {
+    display: flex;
+    align-items: center;
+  }
+
+  &.Mui-disabled {
+    color: inherit;
+  }
+`;
 
-Legend.defaultProps = {
-  isExporting: false,
+const Box = styled.span`
+  display: block;
+  width: 1.25rem;
+  height: 1.25rem;
+  margin-right: 0.625rem;
+  border-radius: 3px;
+`;
+
+const Text = styled.span`
+  line-height: 1.4;
+`;
+
+const getDisplayValue = (chartConfig, value) => {
+  if (value in chartConfig && 'label' in chartConfig[value]) {
+    return chartConfig[value].label;
+  }
+  return value;
 };
+
+export const getPieLegend = () => ({ payload }) => (
+  <LegendContainer style={{ padding: 0, marginBottom: -10 }}>
+    {payload.map(({ color, value }) => (
+      <LegendItem key={value} disabled>
+        <Box style={{ background: color }} />
+        <Text>{value}</Text>
```

Comment:
We don't want `getDisplayValue` here?

Reply:
We don't support this on tupaia yet but might as well since we are here.